chroma_core/lib/job.py

```diff
@@ -200,7 +200,21 @@ def invoke_rust_agent(self, host, command, args={}):
     def invoke_rust_agent_expect_result(self, host, command, args={}):
         from chroma_core.services.job_scheduler.agent_rpc import AgentException
 
-        result = json.loads(self.invoke_rust_agent(host, command, args))
+        try:
+            result = self.invoke_rust_agent(host, command, args)
+        except RustAgentCancellation as e:
+            raise AgentException(host, command, args, "Cancelled: {}".format(e))
+        except ValueError as e:
```

Comment:
We definitely need the ValueError below when parsing json but i'm wondering if I need it here as well?

Reply:
Nope.

Reply:
ok great, i'll remove then.

Reply:
To be clear, we shouldn't need the `ValueError` case.